weatherApp/Features/SearchLocation/UI/LocationSearchViewController.swift

```diff
@@ -0,0 +1,190 @@
+//
+//  LocationSearchViewController.swift
+//  weatherApp
+//
+//  Created by Dario Nikolic on 03/09/2020.
+//  Copyright © 2020 Dario Nikolic. All rights reserved.
+//
+
+import UIKit
+import RxSwift
+import RxCocoa
+
+protocol LocationSearchDelegate {
+    func didTapNewLocation()
+}
+
+class LocationSearchViewController: UIViewController {
+    
+    private let citiesDisposeBage: DisposeBag = DisposeBag()
+    private let searchBarDisposeBag: DisposeBag = DisposeBag()
+    private var locationSearchDelegate: LocationSearchDelegate?
+    private var presenter: LocationSearchPresenter!
+    private let citiesFiltered: BehaviorRelay<[CityViewModel]> = BehaviorRelay<[CityViewModel]>(value: [])
+    private var locationSearchView: LocationSearchView {
+        return self.view as! LocationSearchView
+    }
+    
+    convenience init(with presenter: LocationSearchPresenter) {
+        self.init()
+        self.presenter = presenter
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setupSearchBar()
+        setupKeyboardDismissGestureRecognizer()
+        setupBackButton()
+        setupTableView()
+        setupDataFiltering()
+        setupTableViewDataReloading()
+    }
+    
+    override func loadView() {
+        self.view = LocationSearchView(frame: UIScreen.main.bounds)
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        navigationController?.setNavigationBarHidden(true, animated: animated)
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        navigationController?.setNavigationBarHidden(false, animated: animated)
+    }
+    
+    func setupDelegate(_ delegate: LocationSearchDelegate) {
+        locationSearchDelegate = delegate
+    }
+    
+    private func setupSearchBar() {
+        setupSearchBarAnimation()
+        setupSearchButtonInteractions()
+        locationSearchView.searchBar.becomeFirstResponder()
+    }
+    
+    private func setupSearchBarAnimation() {
+        locationSearchView.searchBar.rx
```

Comment:
If you want you can use .combineLatest for these two observables.

Reply:
Tried to implement with combining observables but it turns out rx.textBeginEditin and rx.txtDidEndEditing return Event<Void> so i wouldn't know which Event was fired up if i get latestValues for both